march_rqt_gait_selection/src/march_rqt_gait_selection/gait_selection_view.py

```diff
@@ -0,0 +1,259 @@
+        self._is_refresh_active = True
+        self._clear_gui(clear_gait_menu=True)
+
+        self.available_gaits = self._controller.get_directory_structure()
+        self.version_map = self._controller.get_version_map()
+
+        self._gait_menu.addItems(sorted(self.available_gaits.keys()))
+
+        self._log('Directory data refreshed', 'success')
+        self._is_refresh_active = False
+
+    # logger
+    def _log(self, msg, color_tag='info'):
+        """Use the logger window in the GUI to display data.
+
+        :param msg:
+            The message to display in the widget
+        :param color_tag:
+            The tag which represents the color of the text in the screen (info, warning, error)
+        """
+        try:
+            color = self._colors[color_tag]
+        except KeyError:
+            color = self._colors['error']
+
+        self._logger.appendHtml('<p style="color:' + str(color) + '">' + msg + '</p>')
+        scrollbar = self.findChild(QPlainTextEdit, 'Log').verticalScrollBar()
+        scrollbar.setValue(scrollbar.maximum())
+
+    # button functions
+    def _clear_gui(self, clear_gait_menu=False):
+        """Clear the subgait menus and subgait labels and optionally the gait menu.
+
+        :param clear_gait_menu:
+            Set this to true if the gait menu should also be cleared
+        """
+        for subgait_menu in self._subgait_menus:
+            subgait_menu.clear()
+
+        for subgait_label in self._subgait_labels:
+            subgait_label.setStyleSheet('color:#000000')
+            subgait_label.setText('Unused')
+
+        if clear_gait_menu:
+            self._gait_menu.clear()
+
+    def _save_default(self):
+        """Save the currently selected subgait versions as default values."""
+        if self._controller.set_default_versions():
+            self._log('Set new default versions', 'success')
+        else:
+            self._log('Set new default versions failed', 'error')
+
+    def _apply(self):
+        """Apply newly selected subgait versions to the gait selection node."""
+        self.update_selected_versions()
+        if self._controller.set_version_map(self.version_map):
+            self._log('Version change applied', 'success')
+        else:
+            self._log('Version change applied failed', 'failed')
+
+    def _show_version_map_pop_up(self):
+        """Use a pop up window to display all the gait, subgaits and currently used versions."""
+        version_map = self._controller.get_version_map()
+        version_map_string = ''
+
+        for gait_name in sorted(version_map.keys()):
+            version_map_string += '{gait} \n'.format(gait=gait_name)
+            for subgait_name, version in version_map[gait_name].items():
+                version_map_string += '\t{sb:<30} \t {vs} \n'.format(sb=subgait_name, vs=version)
+            version_map_string += '\n'
+
+        self._version_map_pop_up.show_message(version_map_string)
+
+
+class PopUpWindow(QDialog):
```

Comment:
Perhaps move this class to a separate file?

Reply:
Thought about this as well but I hesitated because the class is so small. But it is indeed more conventional to separate this from the gait_selection_view.py.